### PR TITLE
enh(UI): add button CSV to export graph datas csv file

### PR DESCRIPTION
**Pull request by Jordan91**
## Description

As a user of the Centreon solution, I would like to be able to do a CSV export from my data in the graph of the resource status menu in order to make a custom use of my supervision data.

## Type of change

- [x] New functionality (non-breaking change)

## Target serie

- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

exportGraph

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/front_src/src/Resources/Details/index.test.tsx [hunk 1]
```diff
@@ -1801,4 +1806,35 @@ describe(Details, () => {
       expect(getByText(email)).toBeInTheDocument();
     });
   });
+
+  it('calls the download graph endpoint when the Graph tab is selected and the "CSV label" is clicked', async () => {
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/index.test.tsx]:
```suggestion
  it('calls the download graph endpoint when the Graph tab is selected and the "Export to CSV button" is clicked', async () => {
```

File: www/front_src/src/Resources/translatedLabels.ts [hunk 2]
```diff
@@ -257,3 +256,8 @@ export const labelServiceSeverity = 'Service severity';
 export const labelHostSeverity = 'Host severity';
 export const labelHostSeverityLevel = 'Host severity level';
 export const labelServiceSeverityLevel = 'Service severity level';
+export const labelExportToCSV = 'CSV';
+export const labelExport = 'Export';
+export const labelAsDisplayed = 'PNG as displayed';
+export const labelAsMediumSize = 'PNG as Medium size';
+export const labelAsSmallSize = 'PNG as Small size';
```
Comment (Thebarda) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
export const labelAsDisplayed = 'Image as displayed';
export const labelAsMediumSize = 'Image as medium size';
export const labelAsSmallSize = 'Image as small size';
```

**Review by Thebarda — COMMENTED** [review]
Don't forget to add translations

File: www/front_src/src/Resources/translatedLabels.ts (hunk 2 above)
Comment (Jordan91) [reply www/front_src/src/Resources/translatedLabels.ts]: Thansk's for your review

File: www/front_src/src/Resources/translatedLabels.ts [hunk 3]
```diff
@@ -257,3 +256,8 @@ export const labelServiceSeverity = 'Service severity';
 export const labelHostSeverity = 'Host severity';
 export const labelHostSeverityLevel = 'Host severity level';
 export const labelServiceSeverityLevel = 'Service severity level';
+export const labelExportToCSV = 'CSV';
+export const labelExport = 'Export';
+export const labelAsDisplayed = 'Image as displayed';
+export const labelAsMediumSize = 'Image as Medium size';
+export const labelAsSmallSize = 'Image as Small size';
```
Comment (Thebarda) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
export const labelAsMediumSize = 'Image as medium size';
export const labelAsSmallSize = 'Image as small size';
```

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -15220,4 +15220,13 @@ msgstr "Nivel de criticidad del host"
...
+#  msgstr ""
+
+# msgid "Image as displayed"
+# msgstr "Image as displayed"
+
+# msgid "Image as Medium size"
+# msgstr "Image as Medium size"
+
+# msgid "Image as Small size"
+# msgstr "Image as Small size"
```
Comment (Thebarda) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
# msgid "Image as displayed"
# msgstr ""

# msgid "Image as medium size"
# msgstr ""

# msgid "Image as small size"
# msgstr ""
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -17013,3 +17013,12 @@ msgstr "Chemin du certificat de l’autorité de certification (optionnel)"
...
 msgid "Certificate Common Name (optional)"
 msgstr "\"Common Name\" du certificat du serveur (optionnel)"
+
+msgid "Image as displayed"
+msgstr "Image affichée"
+
+msgid "Image as Medium size"
+msgstr "Image en taille moyenne"
+
+msgid "Image as Small size"
```
Comment (Thebarda) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Image as medium size"
msgstr "Image en taille moyenne"

msgid "Image as small size"
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -15671,4 +15671,13 @@ msgstr "Nível de Severidade de host"
...
+#  msgstr ""
+
+# msgid "Image as displayed"
+# msgstr "Image as displayed"
+
+# msgid "Image as Medium size"
+# msgstr "Image as Medium size"
+
+# msgid "Image as Small size"
+# msgstr "Image as Small size"
```
Comment (Thebarda) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
# msgid "Image as displayed"
# msgstr "Image as displayed"

# msgid "Image as medium size"
# msgstr ""

# msgid "Image as small size"
# msgstr ""
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -15659,4 +15659,13 @@ msgstr "Nível de Severidade de host"
...
+#  msgstr ""
+
+# msgid "Image as displayed"
+# msgstr "Image as displayed"
+
+# msgid "Image as Medium size"
+# msgstr "Image as Medium size"
+
+# msgid "Image as Small size"
+# msgstr "Image as Small size"
```
Comment (Thebarda) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
# msgid "Image as displayed"
# msgstr ""

# msgid "Image as medium size"
# msgstr ""

# msgid "Image as small size"
# msgstr ""
```

File: www/front_src/src/Resources/Graph/Performance/GraphActions.tsx [hunk 8]
```diff
@@ -71,6 +75,14 @@ const GraphActions = ({
   const closeSizeExportMenu = (): void => {
     setMenuAnchor(null);
   };
+
+  const details = useAtomValue(detailsAtom);
+  const graphToCsvEndpoint = `${details?.links.endpoints.performance_graph}/download?start_date=2022-08-01T00:00:22Z&end_date=2022-08-08T18:00:22Z`;
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/GraphActions.tsx]: Why are the dates hardcoded here?

Comment (Jordan91) [reply www/front_src/src/Resources/Graph/Performance/GraphActions.tsx]: an omission

**Comment by bdauria** [comment]
@Jordan91 
There is an UI issue with what I'm seeing on the screenshot. The title 'Export' seems to be related to the PNG buttons, whereas in fact, it should be global to all the items. 
What's even more weird is that you get a button with the exact same style ('CSV'), which is clickable and the other is not. This should be corrected

**Review by bdauria — DISMISSED** [review]

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -17013,3 +17013,12 @@ msgstr "Chemin du certificat de l’autorité de certification (optionnel)"
 
 msgid "Certificate Common Name (optional)"
 msgstr "\"Common Name\" du certificat du serveur (optionnel)"
+
+msgid "Image as displayed"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "As displayed"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 10]
```diff
@@ -17013,3 +17013,12 @@ msgstr "Chemin du certificat de l’autorité de certification (optionnel)"
...
+msgstr "Image affichée"
+
+msgid "Image as medium size"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Medium size"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 11]
```diff
@@ -17013,3 +17013,12 @@ msgstr "Chemin du certificat de l’autorité de certification (optionnel)"
...
+msgstr "Image en taille moyenne"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Taille moyenne"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 12]
```diff
@@ -17013,3 +17013,12 @@ msgstr "Chemin du certificat de l’autorité de certification (optionnel)"
...
+
+msgid "Image as small size"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Small size"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 13]
```diff
@@ -17013,3 +17013,12 @@ msgstr "Chemin du certificat de l’autorité de certification (optionnel)"
...
+msgstr "Image en petite taille"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Petite taille"
```

File: www/front_src/src/Resources/translatedLabels.ts [hunk 14]
```diff
@@ -257,3 +256,8 @@ export const labelServiceSeverity = 'Service severity';
 export const labelHostSeverity = 'Host severity';
 export const labelHostSeverityLevel = 'Host severity level';
 export const labelServiceSeverityLevel = 'Service severity level';
+export const labelExportToCSV = 'CSV';
+export const labelExport = 'Export';
+export const labelAsDisplayed = 'Image as displayed';
```
Comment (cg-tw) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
export const labelAsDisplayed = 'As displayed';
```

File: www/front_src/src/Resources/translatedLabels.ts [hunk 15]
```diff
@@ -257,3 +256,8 @@ export const labelServiceSeverity = 'Service severity';
...
+export const labelAsMediumSize = 'Image as medium size';
```
Comment (cg-tw) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
export const labelAsMediumSize = 'Medium size';
```

File: www/front_src/src/Resources/translatedLabels.ts [hunk 16]
```diff
@@ -257,3 +256,8 @@ export const labelServiceSeverity = 'Service severity';
...
+export const labelAsSmallSize = 'Image as small size';
```
Comment (cg-tw) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
export const labelAsSmallSize = 'Small size';
```

Comment (Thebarda) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
export const labelAsDisplayed = 'As displayed';
export const labelAsMediumSize = 'Medium size';
export const labelAsSmallSize = 'Small size';
```

**Review by Thebarda — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Pull request merged**